src/bin/lpython.cpp

```diff
@@ -1616,7 +1635,7 @@ int main(int argc, char *argv[])
         app.require_subcommand(0, 1);
         CLI11_PARSE(app, argc, argv);
 
-        lcompilers_unique_ID = LCompilers::get_unique_ID();
+        lcompilers_unique_ID = separate_compilation ? LCompilers::get_unique_ID(): "";
```

Comment:
I would not do that here, but rather have `lcompilers_unique_ID` always have a unique ID. But have this logic of turning it on and off in asr_scopes. If asr_scopes doesn't have access to `separate_compilation` then let's create another variable `lcompilers_unique_ID_separate_compilation` and have that variable on and off as needed.